torch/distributed/_shard/api.py

```diff
@@ -244,6 +252,11 @@ def shard_module(
 
     Needs to be called on all ranks in an SPMD fashion.
 
+    If the parameter is a MetaTensor, then an empty ShardedTensor is
+    materialized as per the provided ``sharding_spec`` and ``src_rank`` is ignored.
```

Comment:
We should switch `materialized` with created or initialized.

Materialize is how we call converting a meta tensor into a tensor using a real device.